hammr/commands/scan/scan.py

```diff
@@ -493,10 +469,13 @@ def help_delete(self):
     def upload_and_launch_scan_binary(self, uforge_login, uforge_password, uforge_apikeys, args, file_src_path, uforge_url):
         hostname = args.ip
         username = args.login
+        id_file = args.id_file
+        password = None
 
-        if not args.password:
-            password = getpass.getpass('Password for %s@%s: ' % (username, hostname))
-        else:
+        if not id_file:
+            if not args.password:
+                password = getpass.getpass('Password for %s@%s: ' % (username, hostname))
+        if args.password:
             password = args.password
```

Comment:
```
if not args.password:
                password = getpass.getpass('Password for %s@%s: ' % (username, hostname))
else:
                password = args.password
```
Is it not better that way? One checked removed

Reply:
but in this case you do not handle the case where you give an identity file (ssh key) and you don't need to ask for a password

Reply:
Yes you are right ;)